jill/utils/version_utils.py

```diff
@@ -206,125 +143,26 @@ def latest_version(version: str, system, architecture, update=True, **kwargs) ->
     if is_full_version(version):
         return version
 
-    if update:
-        print(f"query the latest {version} version, it may take seconds...")
-    if len(version.strip()) == 0:
-        # if empty string is provided, query the latest version since the latest stable released
-        # version. This version might not be released yet for specific platform, e.g., ARMv7, in
-        # this case, we query the database and return the latest one
-        versions = sorted(
-            set(map(lambda x: x[0], read_releases(stable_only=True))), key=Version)
-        ver = latest_major_version(
-            max(versions), system, architecture, update=update, **kwargs)
+    # query system/architecture-compatible releases
+    releases = read_releases(stable_only=stable_only)
+    compat_releases = list(filter(lambda x: x[1] == system and x[2] == architecture, releases))
+    if len(compat_releases) == 0:
+        latest_ver = max(releases, key=lambda x: Version(x[0]))[0]
+        print(
+            f'{color.RED}failed to find latest Julia version for "{system}" and "{architecture}". Trying latest compatible version "{latest_ver}" instead.{color.END}')
+        return latest_ver
```

Comment:
What's the reasoning behind this? My understanding is, if `compat_releases` is an empty set then it's unlikely to download this `(latest_ver, sys, arch)` release in any of the upstream.

Reply:
Makes sense. I think I was influenced by the old code and merely trying to translate and simplify it.